single-line-diagram/single-line-diagram-core/src/main/java/com/powsybl/sld/svg/DefaultSVGWriter.java

```diff
@@ -859,6 +860,13 @@ protected void insertFeederInfos(String prefixId,
 
         double shiftFeederInfo = 0;
         for (FeederInfo feederInfo : labelProvider.getFeederInfos(feederNode)) {
+            String voltageLevelId = graph.getVoltageLevelInfos().getId();
+            if (feederNode.getFeeder() instanceof FeederTwLeg feederTwLeg) {
+                String feederTwLegVoltageLevelId = feederTwLeg.getVoltageLevelInfos().getId();
+                if (!voltageLevelId.equals(feederTwLegVoltageLevelId) && feederInfo instanceof DirectionalFeederInfo directionalFeederInfo) {
+                    directionalFeederInfo.reverseArrowDirectionAndValue();
```

Comment:
I think this reverse should be done within the label provider. If a user of powsybl-diagram was using its own labelProvider you're reversing his results.

It would also be better to do it at the FeederInfo construction, not afterwards.

Reply:
Agreed, I will look for a way to do that without breaking the API.